File: apps/mobile/lib/features/map/map_status.dart
import 'package:flutter/material.dart';

// Derived "map status" colour buckets for the cadastral map — the exact
// product-mandated palette mirrored from the web app
// (apps/web/src/app/shared/map-status.ts). The server computes `map_status`
// authoritatively; the client only paints it.
//
//   🟢 clear     — deed issued, no encumbrance, privately owned → transferable
//   🔴 disputed  — active court seizure / lien / waqf → no dealing allowed
//   🟡 pending   — still in the registration workflow
//   🔵 public    — deed issued, governmental / public-utility ownership
class MapStatusMeta {
  final String ar;
  final Color color;
  final String hint;
  const MapStatusMeta(this.ar, this.color, this.hint);
}

const Map<String, MapStatusMeta> kMapStatus = {
  'clear': MapStatusMeta(
    'سليم — قابل للتصرّف',
    Color(0xFF16A34A),
    'العقار معتمد وخالٍ من النزاعات.',
  ),
  'disputed': MapStatusMeta(
    'نزاع / حجز قانوني',
    Color(0xFFDC2626),
    'عليه نزاع قضائي أو حجز أو منع تصرّف.',
  ),
  'pending': MapStatusMeta(
    'قيد المراجعة',
    Color(0xFFF59E0B),
    'لم يكتمل التحقّق أو ما زال طلب التسجيل قائماً.',
  ),
  'public': MapStatusMeta(
    'ملكية عامة',
    Color(0xFF2563EB),
    'ملكية حكومية أو مرفق عام.',
  ),
};

MapStatusMeta mapStatusMeta(String s) => kMapStatus[s] ?? kMapStatus['pending']!;

const List<String> kMapStatusOrder = ['clear', 'disputed', 'pending', 'public'];

// Arabic property-type labels (mirrors PROPERTY_TYPE on the web).
const Map<String, String> kPropertyTypeAr = {
  'residential': 'سكني',
  'agricultural': 'زراعي',
  'commercial': 'تجاري',
  'governmental': 'حكومي',
  'industrial': 'صناعي',
  'mixed': 'مختلط',
};

String propertyTypeAr(String t) => kPropertyTypeAr[t] ?? t;

File: apps/web/src/app/shared/map-status.ts
// Derived "map status" colour buckets for the property maps (public +
// officer). These four colours are a deliberate, product-mandated palette for
// the cadastral map ONLY — distinct from the workflow status pills in
// status-pills.ts. The server computes `map_status` authoritatively
// (043_property_map.sql); the client only paints it.
//
//   🟢 clear     — deed issued, no encumbrance, privately owned → transferable
//   🔴 disputed  — active court seizure / lien / waqf → no dealing allowed
//   🟡 pending   — still in the registration workflow (officer map only)
//   🔵 public    — deed issued, governmental / public-utility ownership
export type MapStatus = 'clear' | 'disputed' | 'pending' | 'public';

export interface MapStatusMeta {
  ar: string;
  /** Solid colour for the polygon stroke + marker dot. */
  color: string;
  /** Translucent fill for the polygon interior. */
  fill: string;
  /** One-line meaning shown in the legend. */
  hint: string;
}

export const MAP_STATUS: Record<MapStatus, MapStatusMeta> = {
  clear: {
    ar: 'سليم — قابل للتصرّف',
    color: '#16A34A',
    fill: 'rgba(22, 163, 74, 0.28)',
    hint: 'العقار معتمد وخالٍ من النزاعات.',
  },
  disputed: {
    ar: 'نزاع / حجز قانوني',
    color: '#DC2626',
    fill: 'rgba(220, 38, 38, 0.28)',
    hint: 'عليه نزاع قضائي أو حجز أو منع تصرّف.',
  },
  pending: {
    ar: 'قيد المراجعة',
    color: '#F59E0B',
    fill: 'rgba(245, 158, 11, 0.28)',
    hint: 'لم يكتمل التحقّق أو ما زال طلب التسجيل قائماً.',
  },
  public: {
    ar: 'ملكية عامة',
    color: '#2563EB',
    fill: 'rgba(37, 99, 235, 0.28)',
    hint: 'ملكية حكومية أو مرفق عام.',
  },
};

export function mapStatusMeta(s: string): MapStatusMeta {
  return MAP_STATUS[s as MapStatus] ?? MAP_STATUS.pending;
}

// Order the legend / filter chips are rendered in.
export const MAP_STATUS_ORDER: MapStatus[] = ['clear', 'disputed', 'pending', 'public'];
